formats/generic.go

```diff
@@ -0,0 +1,24 @@
+package formats
+
+import "fmt"
+
+type FormatOptions struct {
+	Strict bool
+}
+
+type FormatHandler interface {
```

Comment:
I would call it DatapointDeserialization since it will be used as a way to represent datapoints coming from the outside into our internal structure and the opposite way. It would also be easy the oposite operation: DatapointSerialization.

What do you think?

Reply:
Right now, only Input is supported, I want the FormatHandler to handle Output as well

Reply:
ok, I'm fine with the fact of having the serialization/derialization in the same struct. 

In terms of naming, I think the functions should be named differently, here is what I propose:
Process could be named Deserialize or Load instead of Process
Output  could be named Serialize or Save instead of Output

Instead of FormatHandler, I would use FormatAdapter which, I think, describes better the job of the interface.